rust/src/applayer.rs

```diff
@@ -58,6 +58,11 @@ pub struct AppLayerTxData {
     /// logger flags for tx logging api
     logged: LoggerFlags,
 
+    /// track file open/logs so we can know how long to keep the tx
+    pub files_opened: u32,
```

Comment:
One thing I don't like is that this increases this per tx structure with 16 bytes (12 for the fields and 4 for padding):
```c
typedef struct {
	AppLayerTxConfig           config;               /*     0     1 */

	/* XXX 3 bytes hole, try to pack */

	LoggerFlags                logged;               /*     4     4 */
	uint32_t                   files_opened;         /*     8     4 */
	uint32_t                   files_logged;         /*    12     4 */
	uint32_t                   files_stored;         /*    16     4 */

	/* XXX 4 bytes hole, try to pack */

	uint64_t                   detect_flags_ts;      /*    24     8 */
	uint64_t                   detect_flags_tc;      /*    32     8 */

	/* size: 40, cachelines: 1, members: 7 */
	/* sum members: 33, holes: 2, sum holes: 7 */
	/* last cacheline: 40 bytes */
} AppLayerTxData;
```
vs master:
```c
typedef struct {
	AppLayerTxConfig           config;               /*     0     1 */

	/* XXX 3 bytes hole, try to pack */

	LoggerFlags                logged;               /*     4     4 */
	uint64_t                   detect_flags_ts;      /*     8     8 */
	uint64_t                   detect_flags_tc;      /*    16     8 */

	/* size: 24, cachelines: 1, members: 4 */
	/* sum members: 21, holes: 1, sum holes: 3 */
	/* last cacheline: 24 bytes */
} AppLayerTxData;
```

Reply:
The reason I've gone with `u32` is that in SMTP (attachments) and HTTP (multipart) we can potentially have a very large number of files per tx.